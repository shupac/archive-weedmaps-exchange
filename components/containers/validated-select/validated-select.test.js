import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import ValidatedForm from 'react-validation/build/form';
import ValidatedSelect from './';

ValidatedForm.displayName = 'ValidatedForm';

describe('validated select', () => {
  it('renders correctly', () => {
    const onChange = () => null;
    const tree = renderer
      .create(
        <ValidatedForm>
          <ValidatedSelect onChange={onChange}>
            <option value="one">One</option>
            <option value="two">Two</option>
            <option value="three">Three</option>
          </ValidatedSelect>
        </ValidatedForm>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onChange when clicking on a SelectionItem', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <ValidatedForm>
        <ValidatedSelect onChange={onChange}>
          <option value="one">One</option>
          <option value="two">Two</option>
          <option value="three">Three</option>
        </ValidatedSelect>
      </ValidatedForm>,
    );
    wrapper.find('SelectionButton').simulate('click');
    wrapper
      .find('SelectionItem')
      .first()
      .simulate('click');
    expect(onChange).toHaveBeenCalled();
  });

  it('should update the form when clicking on a SelectionItem', () => {
    const wrapper = mount(
      <ValidatedForm>
        <ValidatedSelect name="select">
          <option value="one">One</option>
          <option value="two">Two</option>
          <option value="three">Three</option>
        </ValidatedSelect>
      </ValidatedForm>,
    );
    wrapper.find('SelectionButton').simulate('click');
    wrapper
      .find('SelectionItem')
      .first()
      .simulate('click');
    expect(
      wrapper
        .find('ValidatedForm')
        .instance()
        .getValues().select,
    ).toEqual('one');
  });
});
