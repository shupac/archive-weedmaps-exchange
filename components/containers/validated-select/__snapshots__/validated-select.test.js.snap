// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validated select renders correctly 1`] = `
.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 9px;
  height: 9px;
}

.c0 {
  cursor: pointer;
  height: 2.25rem;
  min-width: default;
  background-color: #FFFFFF;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  border: 1px solid #CCCCCC;
  border-radius: 0.1875rem;
  padding-right: 1.875rem;
  padding-left: 0.9375rem;
  padding-top: 0.1875rem;
  font-size: 0.875rem;
  font-weight: 300;
  color: #999999;
  text-align: left;
}

.c0:disabled,
.c0 [disabled] {
  background-color: #CCCCCC;
}

.c1 {
  position: relative;
}

.c2 {
  position: absolute;
  right: -1.25rem;
  bottom: 0.125rem;
}

.c4 {
  position: relative;
}

<form>
  <div>
    <button
      aria-expanded={false}
      aria-haspopup={true}
      aria-label="open menu"
      className="c0"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      type="button"
    >
      <div
        className="c1"
      >
        <div
          className="c2"
        >
          <div
            className="c3"
            size={
              Object {
                "height": "9px",
                "width": "9px",
              }
            }
          >
            <svg
              height="9px"
              viewBox="0 0 10 6"
              width="9px"
            >
              <defs>
                <path
                  d="M2 2.998L5 8h2.875L4.856 2.998 8-2H5z"
                  id="a"
                />
              </defs>
              <g
                fill="none"
                fillRule="evenodd"
                transform="translate(-1421 -27)"
              >
                <g
                  transform="translate(197)"
                >
                  <g
                    transform="translate(1103 10)"
                  >
                    <g
                      transform="translate(121 17)"
                    >
                      <path
                        d="M2 2.998L5 8h2.875L4.856 2.998 8-2H5z"
                        fill="#999999"
                        transform="rotate(-90 5 3)"
                      />
                    </g>
                  </g>
                </g>
              </g>
            </svg>
          </div>
        </div>
      </div>
    </button>
    <div
      className="c4"
    />
  </div>
</form>
`;
